lib/dnsimple/error.rb

```diff
@@ -7,15 +7,25 @@ class Error < StandardError
 
   # RequestError is raised when an API request fails for an client, a server error or invalid request information.
   class RequestError < Error
-    attr_reader :http_response
+    attr_reader :http_response, :errors
 
     def initialize(http_response)
       @http_response = http_response
+      @errors = errors_from(http_response)
       super(message_from(http_response))
     end
 
     private
 
+    def errors_from(http_response)
+      content_type = http_response.headers["Content-Type"]
+      if content_type&.start_with?("application/json")
+        http_response.parsed_response["errors"]
+      else
+        {}
+      end
```

Comment:
```suggestion
      content_type = http_response.headers["Content-Type"]

      return unless content_type&.start_with?("application/json")

      http_response.parsed_response["errors"]
```

Reply:
I wonder why people hate explicit `if-else` so much. It's all a matter of personal preference, but to me, it is so much more obvious at first sight what's going on...

Reply:
I don't think it's hate, it's more likely they are looking for terseness over legibility.

I have a different question about the suggestion though: aren't they different behaviors? The original option will return an empty hash if the content type doesn't start with "application/json" whereas the second version returns nil, or am I misreading this? Is there an expectation one way or another about what the return value will be?

Reply:
> I don't think it's hate, it's more likely they are looking for terseness over legibility.

To me, readability is the most important thing once the code does what it's supposed to do.

> The original option will return an empty hash if the content type doesn't start with "application/json" whereas the second version returns nil, or am I misreading this? Is there an expectation one way or another about what the return value will be?

That's correct. It also leaves me wondering if the suggestion is about the flow of the code, the return value or both. 🤔 

We should be consistent with other clients, so most likely `nil` is the better option. 

Thanks for pointing this out @aeden.

Reply:
☝️ I am going to add an explicit test scenario for what happens with errors responses without an `errors` entry, which is actually the best indicator for the value being `nil`.

Reply:
I have updated the code and re-requested review. Thanks for your input @san983

Reply:
My suggestion was about both. And you got me right by adding that extra spec! Double ✅